python/src/aac/lang/definitions/schema.py

```diff
@@ -94,8 +102,72 @@ def _get_sub_definitions(schema_definition: Definition, fields):
+                    _get_sub_definitions(
+                        field_definition, field_definition.get_top_level_fields().get(DEFINITION_FIELD_FIELDS)
+                    )
 
     top_level_fields = list(get_schema_defined_fields(source_definition, context).values())
     _get_sub_definitions(source_definition, top_level_fields)
     return list(substructure_definitions.values())
+
+
+def get_schema_for_field(
+    source_definition: Definition, field_keys: list[str], context: LanguageContext
+) -> Optional[Definition]:
+    """
+    Returns the schema definition for a field using a list of field names (keys) to target the field.
+
+    For example, if you wanted to know the definition for ['model','component'] you'd get 'Field' because components is an array of `Field`.
+
+    Args:
+        source_definition (Definition): The definition to search through
+        field_keys (list[str]): The list of keys used to identify the target field in the source definition
+        context (LanguageContext): The language context, used to navigate the structure and lookup definitions
+
+    Returns:
+        The schema that defines the field structure, the enum definition defined for an enum field, or the primitives definition for primitive fields.
+    """
+    keys_to_traverse = field_keys.copy()
+    definition_to_return = None
+
+    def _traverse_key(fields: dict) -> Optional[Definition]:
+        field_schema_definition = None
+
+        key_to_traverse = keys_to_traverse.pop(0)
+        field_to_traverse = fields.get(key_to_traverse, {})
+
+        field_type = str(field_to_traverse.get(DEFINITION_FIELD_TYPE, ""))
+
+        if field_type:
+            if context.is_definition_type(field_type):
+                field_schema_definition = context.get_definition_by_name(field_type)
+                fields_to_traverse = field_schema_definition.get_top_level_fields().get(DEFINITION_FIELD_FIELDS, {})
+                traverse_fields_dict = _convert_fields_list_to_dict(fields_to_traverse)
+
+                if len(keys_to_traverse) > 0 and field_schema_definition:
+                    field_schema_definition = _traverse_key(traverse_fields_dict)
+
+            elif context.is_primitive_type(field_type):
+                field_schema_definition = context.get_definition_by_name(DEFINITION_NAME_PRIMITIVES)
+            else:
+                # Assumed to be enum as enums and primitives are the only terminal types
+                field_schema_definition = context.get_enum_definition_by_type(field_type)
+
+        else:
+            # In the case of no field-type, we're assumed to be at a terminal value (enum or primitive value)
+            field_schema_definition = context.get_enum_definition_by_type(key_to_traverse)
+
+        return field_schema_definition
+
+    if len(keys_to_traverse) > 1:
+        keys_to_traverse.pop(0)  # Go ahead and pop the root key since we don't specifically need it.
```

Comment:
Why have users provide the root key if we don't need it?

Reply:
You do need it on line 165, just not in the recursive calls.